Sources/PointFree/Stripe.swift

```diff
@@ -112,29 +127,59 @@ public struct Stripe {
 
   public struct Invoice: Codable {
     public private(set) var amountDue: Cents
+    public private(set) var amountPaid: Cents
+    public private(set) var charge: Charge?
+    public private(set) var closed: Bool
     public private(set) var customer: Customer.Id
+    public private(set) var date: Date
     public private(set) var id: Id
-    public private(set) var subscription: Subscription.Id
+    public private(set) var lines: ListEnvelope<LineItem>
+    public private(set) var number: Number
+    public private(set) var periodStart: Date
+    public private(set) var periodEnd: Date
+    public private(set) var subscription: Subscription.Id?
+    public private(set) var subtotal: Cents
+    public private(set) var total: Cents
 
-    public typealias Id = Tagged<Invoice, String>
+    public typealias Id = Tagged<(Invoice, id: ()), String>
+    public typealias Number = Tagged<(Invoice, number: ()), String>
```

Comment:
Turns out we don't need to create one-off types for tag collisions! These two types aren't considered equal! 😄

Reply:
lol amazing!